tidy3d/plugins/dispersion/fit.py

```diff
@@ -448,9 +489,144 @@ def plot(
 
         return ax
 
+    @staticmethod
+    def validate_url_load(data_load: List):
+        """Validate if the loaded data from URL is valid
+            The data list should be in this format:
+                [["wl",     "n"],
+                 [float,  float],
+                  .        .
+                  .        .
+                  .        .
+            (if lossy)
+                 ["wl",     "k"],
+                 [float,  float],
+                  .        .
+                  .        .
+                  .        .]]
+
+        Parameters
+        ----------
+        data_load : List
+            Loaded data from URL
+
+        Raises
+        ------
+        ValidationError
+            Or other exceptions
+        """
+        has_k = 0
+
+        if data_load[0][0] != "wl" or data_load[0][1] != "n":
+            raise ValidationError(
+                "Invalid URL. The file should begin with ['wl','n']. "
+                "Or make sure that you have supplied an appropriate delimiter."
+            )
+
+        for row in data_load[1:]:
+            if row[0] == "wl":
+                if row[1] == "k":
+                    has_k += 1
+                else:
+                    raise ValidationError(
+                        "Invalid URL. The file is not well formatted for ['wl', 'k'] data."
+                    )
+            else:
+                # make sure the rest is float type
+                nk_tmp = [float(x) for x in row]  # pylint:disable=unused-variable
+
+        if has_k > 1:
+            raise ValidationError("Invalid URL. Too many k labels.")
+
+    @classmethod
+    def from_url(cls, url_file: str, delimiter=","):
+        """loads ``DispersionFitter`` from url pointing to a csv/txt file
+            that contains wavelength (micron), n, and optionally k data.
+            Preferred from refractiveindex.info
+
+            The data should be in this format:
+                wl     n
+                float  float
+                .      .
+                .      .
+                .      .
+            (if lossy)
+                wl     k
+                float  float
+                .      .
+                .      .
+                .      .
+
+            The delimiter can be either ',' (csv file), or '\t' (txt file)
+
+        Parameters
+        ----------
+        url_file : str
+            Url link to the csv file.
+            e.g. "https://refractiveindex.info/data_csv.php?datafile=data/main/Ag/Johnson.yml"
+        delimiter : str, optional
```

Comment:
instead of optional, lets do `str = ","`

Reply:
resolved